### PR TITLE
Add utility function for tracking function calls

### PR DESCRIPTION
**Pull request by yyc**
add a global `__track_function`, along with `__reset_function_count` and `__get_function_count` to track the use of functions while autograding.

Also add test cases.

Note that the tracking function has to be called in the globals initialization, in order for the function to be wrapped in time. (see test_utils.ts:17)

### CODE REVIEW

File: src/__tests__/test_utils.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    name: 'NONE',
+    symbols: [
+      '__track_function',
+      '__reset_function_count',
+      '__get_function_count',
+      'make_funny',
+    ]
+  },
+  globals: [
+    ['__tracking_function', 'global.__track_function("make_funny");'],
```
Comment (remo5000) [inline src/__tests__/test_utils.ts]: This global doesn't seem to be used, is it required?

File: src/util.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/**
+  Utility function that wraps a function and provides two tracking functions
+  to allow you to track whether a function has been called
+  Type: (fn) -> fn
+**/
+var table = {};
```
Comment (remo5000) [inline src/util.js]: Seeing as how this isn't huge and archaic (like graphics), could it be written in Typescript instead?

File: src/util.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+// Used in testing
+function make_funny(str) {
+  return str + ", haha!";
+}
+
+global.__track_function = track_function;
+global.__reset_function_count = reset_count;
+global.__get_function_count = get_count;
+global.make_funny = make_funny;
```
Comment (remo5000) [inline src/util.js]: We shouldn't be exposing this test function in the global scope every time the `runAll` function is called, could you expose this in your tests using `globals`?

File: src/__tests__/test_utils.ts (hunk 1 above)
Comment (yyc) [reply src/__tests__/test_utils.ts]: We just need the __track_function to be evaluated during the globals initialization. The actual value isn't used.

File: src/util.js (hunk 3 above)
Comment (yyc) [reply src/util.js]: good call. changed.

File: src/util.js (hunk 2 above)
Comment (yyc) [reply src/util.js]: done

**Pull request merged**